engine/src/main/java/net/jqwik/engine/support/types/TypeUsageImpl.java

```diff
@@ -661,6 +689,14 @@ public List<TypeUsage> getInterfaces() {
 		return toTypeUsages(getRawType().getGenericInterfaces());
 	}
 
+	@Override
+	public List<TypeUsage> getSuperTypes() {
+		List<TypeUsage> supertypes = new ArrayList<>();
+		getSuperclass().ifPresent(supertypes::add);
```

Comment:
This seems to retrieve at most one super type rather than all of them

Reply:
There can only be one superclass. What am I missing?
It's only about the _direct_ supertypes not indirect ones. Those are being picked when diving deeper.

Reply:
In that case `getDirectSuperTypes` would make the code easier to follow

Reply:
Since it's also `getSuperclass/getInterfaces` and not `getDirectSuperclass` that would be inconsistent IMO.